internal/incident/sync.go

```diff
@@ -150,23 +154,28 @@ func (i *Incident) addPendingNotifications(
 				Type:              Notified,
 				ChannelID:         utils.ToDBInt(chID),
 				NotificationState: NotificationStatePending,
+				Message:           utils.ToDBString(ev.Message),
+			}
+			if i.isMuted && i.Object.IsMuted() {
+				hr.NotificationState = NotificationSateSuppressed
 			}
 
 			if err := hr.Sync(ctx, i.db, tx); err != nil {
-				i.logger.Errorw(
-					"Failed to insert contact pending notification incident history",
-					zap.String("contact", contact.String()), zap.Error(err),
-				)
+				i.logger.Errorw("Failed to insert incident notification history",
+					zap.String("contact", contact.FullName), zap.Bool("incident_muted", i.Object.IsMuted()),
+					zap.Error(err))
 
-				return nil, errors.New("can't insert contact pending notification incident history")
+				return nil, errors.New("cannot insert incident notification history")
 			}
 
-			notifications = append(notifications, &NotificationEntry{
-				HistoryRowID: hr.ID,
-				ContactID:    contact.ID,
-				State:        NotificationStatePending,
-				ChannelID:    chID,
-			})
+			if !i.isMuted || !i.Object.IsMuted() {
```

Comment:
I'd either do a `suppress := i.isMuted && i.Object.IsMuted()` at the beging of the function and use it for both `if`s, or use `hr.State == NotificationStatePending` as the condition for the last `if`.